source/MaterialXTest/MaterialXCore/Material.cpp

```diff
@@ -54,3 +58,143 @@ TEST_CASE("Material", "[material]")
     REQUIRE(instanceSpecColor->getDefaultValue()->asA<mx::Color3>() == mx::Color3(0.0f));
     REQUIRE(doc->validate());
 }
+
+TEST_CASE("Material Discovery", "[material]")
+{
+    mx::DocumentPtr doc = mx::createDocument();
+
+    mx::FileSearchPath searchPath;
+    const mx::FilePath currentPath = mx::FilePath::getCurrentPath();
+    searchPath.append(currentPath / mx::FilePath("libraries"));
+    searchPath.append(currentPath / mx::FilePath("resources/Materials/TestSuite"));
+
+    searchPath.append(mx::FilePath::getCurrentPath() / mx::FilePath("libraries"));
+    loadLibraries({ "targets", "stdlib", "pbrlib" }, searchPath, doc);
+
+    mx::FilePath filename = "stdlib/materials/material_node_discovery.mtlx";
+    mx::readFromXmlFile(doc, filename, searchPath);
+
+    mx::StringSet foundNames;
+
+    // 1. Find all materials referenced by material assignments.
+    //    Tests both finding direct node connections as well as traversing through
+    //    connected nodegraphs
+    mx::StringSet assignmentMaterialPaths =
+    {
+        "top_level_material_assigned", // reference to material node from standard library
```

Comment:
Could we potentially remove this string set, and just check the count of discovered paths?  It seems ideal to minimize the coupling between unit tests and specific test documents, allowing element names to change over time without our tests falling out of sync.

Any additional simplification that can be made to this unit test would also be appreciated, since its complexity seems out of balance with the simplicity of the newly-added methods.  :)

Reply:
Removed the test on node interfaces + specific names and only kept the ones for outputs. Don't think there is anything simpler now :).